### PR TITLE
Add intersection layer

### PR DESCRIPTION
**Pull request by Algorush**
added intersection Layer with `inersection` component (not automatic version).
As a part of issue: https://github.com/3DStreet/3dstreet/issues/613

### CODE REVIEW

File: src/components/intersection.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+      rotationVec
+    }) {
+      const sd = document.createElement('a-entity');
+      const repeatCountInter = [];
+      repeatCountInter[0] = width / 2;
+      // every 2 meters repeat sidewalk texture
+      repeatCountInter[1] = parseInt(length / 2);
+
+      sd.setAttribute('geometry', 'primitive', 'box');
+      sd.setAttribute('geometry', 'height: 0.4');
```
Comment (vincentfretin) [inline src/components/intersection.js:79]: I know you just moved the code in a separate file, but I'm reading this code for the first time, so making suggestions here.
This should be done in one call.
```js
sd.setAttribute('geometry', 'primitive: box; height: 0.4');
```

File: src/components/intersection.js [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+      sd.setAttribute('position', positionVec);
+      sd.setAttribute('scale', scaleVec);
+      sd.setAttribute('geometry', 'depth', length);
+      sd.setAttribute('geometry', 'width', width);
```
Comment (vincentfretin) [inline src/components/intersection.js:83]:
```js
      sd.setAttribute('geometry', {depth: length, width: width});
```

File: src/components/intersection.js [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    );
+
+    function createSidewalkElem({
+      length,
+      width,
+      positionVec,
+      scaleVec = { x: 1, y: 1, z: 1 },
+      rotationVec
+    }) {
+      const sd = document.createElement('a-entity');
```
Comment (vincentfretin) [inline src/components/intersection.js:72]: Don't we want to add the autocreated class on all those created entities so those are not exported?

File: src/editor/components/components/AddLayerPanel/createLayerFunctions.js [hunk 4]
```diff
@@ -149,11 +149,22 @@ function createPrimitiveGeometry() {
   Events.emit('entitycreated', newEl);
 }
 
+function createIntersection() {
```
Comment (vincentfretin) [inline src/editor/components/components/AddLayerPanel/createLayerFunctions.js:152]: Do you know you can just use `export function createIntersection` here and for all the functions you want to export instead of having an export { ... } at the end of the file?
You don't need to modify that in this PR, I'm just asking if it was a stylistic choice in the code base or if there was any other reason I'm not aware of.

File: src/editor/components/components/AddLayerPanel/layersData.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  createMapbox,
+  createStreetmixStreet,
+  create3DTiles,
+  createCustomModel,
+  createPrimitiveGeometry,
+  createIntersection
+} from './createLayerFunctions';
+
+// data for PRO layers cards
+const layersData = [
```
Comment (vincentfretin) [inline src/editor/components/components/AddLayerPanel/layersData.js:12]: same here you can just use `export const`

File: src/components/intersection.js (hunk 1 above)
Comment (Algorush) [reply src/components/intersection.js:79]: yes I know about this, thanks for reminding

File: src/components/intersection.js (hunk 3 above)
Comment (Algorush) [reply src/components/intersection.js:72]: of course

**Comment by vincentfretin** [comment]
Ok, you can create the PR in draft or convert it to draft now if you're still working on it, and put it on review when it's ready to review.

File: src/editor/components/components/AddLayerPanel/createLayerFunctions.js (hunk 4 above)
Comment (Algorush) [reply src/editor/components/components/AddLayerPanel/createLayerFunctions.js:152]: I stuck to the style that already is. Initially, I don’t know what reasons the programmer had to use this style:)

File: src/editor/components/components/AddLayerPanel/createLayerFunctions.js [hunk 6]
```diff
@@ -149,11 +149,22 @@ function createPrimitiveGeometry() {
...
+  const newEl = document.createElement('a-entity');
+  newEl.setAttribute('intersection', '');
+  newEl.setAttribute('data-layer-name', 'Street intersection');
+  const parentEl = document.querySelector('#street-container');
+  parentEl.appendChild(newEl);
+  // update sceneGraph
+  Events.emit('entitycreated', newEl);
+}
+
```
Comment (vincentfretin) [inline src/editor/components/components/AddLayerPanel/createLayerFunctions.js:161]: To properly implement undo/redo, we need to stop using such code to create an entity. We should use the proper api 
```js
Events.emit('entitycreate', {
  'data-layer-name': 'Street intersection',
  components: {intersection: ''}
});
```
but we need the changes for createEntity from
https://github.com/3DStreet/3dstreet/pull/537/files#diff-2f5a94d2538a12f206cf431af70f02620687da9738381f7be44feeaf809bc5b3R618
We can do that in another PR and modify all the entity creation here.

Comment (Algorush) [reply src/editor/components/components/AddLayerPanel/createLayerFunctions.js:161]: I used `emit('entitycrated')` here to update the scene graph after adding a new element, not to create the entity. But I know that now there is an `updatescenegraph` event, so I'll change to it.

Comment (vincentfretin) [reply src/editor/components/components/AddLayerPanel/createLayerFunctions.js:161]: No no don't change it. entitycreated here is better than updatescenegraph. But we'll need to rewrite all those create functions later that's what I'm saying.

Comment (Algorush) [reply src/editor/components/components/AddLayerPanel/createLayerFunctions.js:161]: Ok, got it

**Comment by Algorush** [comment]
I'm closing this PR because slpat PR (https://github.com/3DStreet/3dstreet/pull/684) is based on this PR.
